pkg/operator/operator.go

```diff
@@ -41,108 +41,17 @@ import (
-
-	DefaultPrometheusQueryInterval                       = time.Minute * 5  // Query Prometheus every 5 minutes
-	DefaultPrometheusQueryStepSize                       = time.Minute      // Query data from Prometheus at a 60 second resolution (one data point per minute max)
-	DefaultPrometheusQueryChunkSize                      = 5 * time.Minute  // the default value for how much data we will insert into Presto per Prometheus query.
-	DefaultPrometheusDataSourceMaxQueryRangeDuration     = 10 * time.Minute // how much data we will query from Prometheus at once
-	DefaultPrometheusDataSourceMaxBackfillImportDuration = 2 * time.Hour    // how far we will query for backlogged data.
+	"github.com/davecgh/go-spew/spew"
+	"github.com/prestodb/presto-go-client/presto"
+	promapi "github.com/prometheus/client_golang/api"
+	prom "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"github.com/taozle/go-hive-driver"
 )
 
-type TLSConfig struct {
-	UseTLS  bool
-	TLSCert string
-	TLSKey  string
-}
-
-func (cfg *TLSConfig) Valid() error {
-	if cfg.UseTLS {
-		if cfg.TLSCert == "" {
-			return fmt.Errorf("Must set TLS certificate if TLS is enabled")
-		}
-		if cfg.TLSKey == "" {
-			return fmt.Errorf("Must set TLS private key if TLS is enabled")
-		}
-	}
-	return nil
-}
-
-type PrometheusConfig struct {
-	Address         string
-	SkipTLSVerify   bool
-	BearerToken     string
-	BearerTokenFile string
-	CAFile          string
-}
-
-type Config struct {
-	Hostname     string
-	OwnNamespace string
-
-	AllNamespaces    bool
-	TargetNamespaces []string
-
-	Kubeconfig string
-
-	APIListen     string
-	MetricsListen string
-	PprofListen   string
-
-	HiveHost                  string
-	HiveUseTLS                bool
-	HiveCAFile                string
-	HiveTLSInsecureSkipVerify bool
-
-	HiveUseClientCertAuth bool
-	HiveClientCertFile    string
-	HiveClientKeyFile     string
-
-	PrestoHost                  string
-	PrestoUseTLS                bool
-	PrestoCAFile                string
-	PrestoTLSInsecureSkipVerify bool
-
-	PrestoUseClientCertAuth bool
-	PrestoClientCertFile    string
-	PrestoClientKeyFile     string
-
-	PrestoMaxQueryLength int
-
-	DisablePrometheusMetricsImporter bool
-	EnableFinalizers                 bool
-
-	LogDMLQueries bool
-	LogDDLQueries bool
-
-	PrometheusQueryConfig                         metering.PrometheusQueryConfig
-	PrometheusDataSourceMaxQueryRangeDuration     time.Duration
-	PrometheusDataSourceMaxBackfillImportDuration time.Duration
-	PrometheusDataSourceGlobalImportFromTime      *time.Time
-
-	ProxyTrustedCABundle string
-
-	LeaderLeaseDuration time.Duration
-
-	APITLSConfig     TLSConfig
-	MetricsTLSConfig TLSConfig
-	PrometheusConfig PrometheusConfig
-}
-
-type Reporting struct {
+// defaultReportingOperator is the concrete implementation of a ReportingOperator.
+type defaultReportingOperator struct {
```

Comment:
Would it work to just call this `reportingOperator` with the lower case indicating its implementation status. I only ask because it's used so many places, it seems like keeping it shorter would be good and yet still code legible. I'm not clear that lower case would do that for Go, but I'm hoping.

Reply:
Lower case variables/functions/methods are for defining global scope throughout a package: https://tour.golang.org/basics/3

So here we're essentially saying you wouldn't have access to this `defaultReportingOperator` type if you're not interacting with the interfaces, instead of the explicit type.

Reply:
if the question is would `ReportingOperator` and `reportingOperator` types conflict the answer is no but I think that's probably too much mental gymnastics.  

I'm ok with shortening the name but didn't have an immediate go to.  I went with `defaultReportingOperator` in the vein of "if we made another one how would we name it?".  Probably something like `instrumentedReportingOperator` or `mockReportingOperator`.  I'm 100% not tied to the name though :smile:

I'll note that referencing it can/should be shorter like we do with `func (op *defaultReportingOperator)` so most of the interactions would simply be `op.Foo` not `defaultReportingOperator.Foo`.

Reply:
Sound reasonable.